marconi/src/Marconi/Index/ScriptTx.hs

```diff
@@ -32,110 +31,265 @@ import Cardano.Ledger.Keys qualified as LedgerShelley
 import Cardano.Ledger.Shelley.Scripts qualified as LedgerShelley
 import Cardano.Ledger.ShelleyMA.Timelocks qualified as Timelock
 
-import RewindableIndex.Index.VSqlite (SqliteIndex)
-import RewindableIndex.Index.VSqlite qualified as Ix
+import RewindableIndex.Storable (Buffered (getStoredEvents, persistToStorage), HasPoint (getPoint),
+                                 QueryInterval (QEverything, QInterval), Queryable (queryStorage),
+                                 Rewindable (rewindStorage), StorableEvent, StorableMonad, StorablePoint, StorableQuery,
+                                 StorableResult, emptyState, filterWithQueryInterval)
+import RewindableIndex.Storable qualified as Storable
 
+{- The first thing that we need to define for a new indexer is the `handler` data
+   type, meant as a wrapper for the connection type (in this case the SQLite
+   connection).
 
-newtype Depth = Depth Int
+   However this is a very good place to add some more configurations
+   that the indexer may require (for example performance tuning settings). In our
+   case we add the number of events that we want to return from the on-disk buffer -}
+
+data ScriptTxHandle = ScriptTxHandle
+  { hdlConnection :: SQL.Connection
```

Comment:
Totally like that we create the connections here rather than just passing in the filepath. thnx